src/lib/Libwin/passwd.c

```diff
@@ -3454,15 +3526,15 @@ get_token_info(HANDLE token,
 
 		if (OpenThreadToken(GetCurrentThread(), TOKEN_ALL_ACCESS,
 			TRUE, &token)  == 0)
-			OpenProcessToken(GetCurrentProcess(), TOKEN_ALL_ACCESS,
-				&token);
+			if (OpenProcessToken(GetCurrentProcess(), TOKEN_ALL_ACCESS,
+					&token) == 0) {
+				log_err(-1, __func__, "failed in OpenProcessToken");
+			}
 
 	}
 
-	if (GetTokenInformation(token, TokenUser, buf, cb, &cb)) {
-		log_err(-1, "get_token_info", "User: 1st GetTokenInformation failed!");
-		return;
-	} else {
+	if ( !GetTokenInformation(token, TokenUser, buf, cb, &cb)) {
+		/* User: 1st GetTokenInformation failed! */
```

Comment:
Because if that fails, we are allocating the buffer and trying GetTokenInformation again, if still that fails then we are logging the corresponding error details

Reply:
I understand that the first call is to query the required buffer lenght. but cant the first GetTokenInformation() fail for reason other than insufficient buffer size?